### PR TITLE
added getnewargs() to str.rs

### PR DESCRIPTION
**Pull request by zer0who**
Added getnewargs() to str.rs and have tested on test_unicode.py(test_getnewargs())

### CODE REVIEW

**Comment by fanninpm** [comment]
#3807 was recently merged, which changes how CI runs the test suites. Can you rebase your branch on a fresh copy of `main`? Once you've done that, feel free to run the following command:

```text
cargo run --release --features ssl,jit -- -m test -j 1 -u all --slowest --fail-env-changed -v test_enum test_pickle test_pickletools test_unicode
```

(Of course, everything after the `-v` flag can be changed out with whatever tests you want. Or none at all, if you have plenty of time to spend.)

**Review by youknowone — COMMENTED** [review]
Great! you fixed many more than just a single test. Could you please remove `expectedFailure` decorators from other successful tests?

File: Lib/test/test_pickle.py [hunk 1]
```diff
@@ -118,13 +118,9 @@ def test_in_band_buffers(self): # TODO: RUSTPYTHON, remove when this passes
     def test_nested_names(self): # TODO: RUSTPYTHON, remove when this passes
         super().test_nested_names() # TODO: RUSTPYTHON, remove when this passes
 
-    # TODO: RUSTPYTHON, AssertionError
-    @unittest.expectedFailure
```
Comment (fanninpm) [inline Lib/test/test_pickle.py:122]: When you delete this, can you also delete the two lines that say `# TODO: RUSTPYTHON, remove when this passes`?

File: Lib/test/test_pickletools.py [hunk 2]
```diff
@@ -40,13 +40,9 @@ def test_in_band_buffers(self): # TODO: RUSTPYTHON, remove when this passes
     def test_nested_names(self): # TODO: RUSTPYTHON, remove when this passes
         super().test_nested_names()
 
-    # TODO: RUSTPYTHON
-    @unittest.expectedFailure
```
Comment (fanninpm) [inline Lib/test/test_pickletools.py:44]: When you delete this, can you also delete the two lines that say `# TODO: RUSTPYTHON, remove when this passes`?

File: Lib/test/test_pickletools.py [hunk 3]
```diff
@@ -40,13 +40,9 @@ def test_in_band_buffers(self): # TODO: RUSTPYTHON, remove when this passes
...
     def test_newobj_generic(self): # TODO: RUSTPYTHON, remove when this passes
         super().test_newobj_generic()
 
-    # TODO: RUSTPYTHON, TypeError: cannot pickle 'weakproxy' object
-    @unittest.expectedFailure
```
Comment (fanninpm) [inline Lib/test/test_pickletools.py:49]: When you delete this, can you also delete the two lines that say `# TODO: RUSTPYTHON, remove when this passes`?

**Review by fanninpm — CHANGES_REQUESTED** [review]
Some minor cleanup comments, and I think this should be good to go.

**Review by fanninpm — CHANGES_REQUESTED** [review]
I apologize if this wasn't clear, but I meant to say the **entirety** of the lines. Right now these lines are just redundant, and they can go away (along with the extra whitespace).

**Comment by fanninpm** [comment]
> You mean that delete the test 'functions' which I deleted expectedFailure annotation?

Any lines that now say
```python
def test_function(self):
    super().test_function()
```
are just redundant at this point if they're not annotated with a `@unittest.expectedFailure` or a `@unittest.skip` decorator. They also make it harder to upgrade the test suite in the future.

If a test function has something else other than a call to `super()`, you can leave it be.

**Comment by zer0who** [comment]
Is this right?

**Comment by zer0who** [comment]
Thank you, too!!

**Comment by fanninpm** [comment]
According to the CI, it seems you may have missed a test in `test_pickle.py`. Are you able to replicate what CI is showing?

**Review by youknowone — APPROVED** [review]
The code changes look good! But maybe you didn't set your git email as your github email.
Do you want to change it before merge?

**Comment by youknowone** [comment]
@zer0who please let me know if you don't care about email address. I don't mind to merge it as it is if you are ok.

**Comment by fanninpm** [comment]
> @zer0who please let me know if you don't care about email address. I don't mind to merge it as it is if you are ok.

[Here is GitHub's documentation to set your commit email address.](https://docs.github.com/en/account-and-profile/setting-up-and-managing-your-personal-account-on-github/managing-email-preferences/setting-your-commit-email-address)

In order to fix these commits, you can run the following commands:

```text
git fetch upstream
git rebase --exec 'git commit --amend --reset-author' -i upstream/main
```

**Comment by zer0who** [comment]
> @zer0who please let me know if you don't care about email address. I don't mind to merge it as it is if you are ok.

I don't know exatly that email you mentioned, but I did it as I understanded. Is it right?

**Comment by youknowone** [comment]
check `git log` in your branch. the email of your commit is not your actual email.
(I don't think your email is `62321708+zer0hoo@users.noreply.github.com`)

**Comment by fanninpm** [comment]
> > @zer0who please let me know if you don't care about email address. I don't mind to merge it as it is if you are ok.
> 
> [Here is GitHub's documentation to set your commit email address.](https://docs.github.com/en/account-and-profile/setting-up-and-managing-your-personal-account-on-github/managing-email-preferences/setting-your-commit-email-address)
> 
> In order to fix these commits, you can run the following commands:
> 
> ```
> git fetch upstream
> git rebase --exec 'git commit --amend --reset-author' -i upstream/main
> ```

I forgot to add `git push --force` to that list.

**Comment by Snowapril** [comment]
@zer0who Yeah It seems good now 😊

**Comment by fanninpm** [comment]
> Am I right??

Here's how you can tell:

- On GitHub, you can look to see if your avatar is displayed next to each of your commits.
- Locally, running [`git log`](https://git-scm.com/docs/git-log) will tell you who authored a given commit. There will normally be a line that says `Author: Your Name <yourname@example.com>`. If the email address in that commit matches one that GitHub recognizes (e.g. the one linked to your GitHub account), then GitHub will display your avatar next to each of your commits.

**Comment by zer0who** [comment]
> > Am I right??
> 
> Here's how you can tell:
> 
> * On GitHub, you can look to see if your avatar is displayed next to each of your commits.
> * Locally, running [`git log`](https://git-scm.com/docs/git-log) will tell you who authored a given commit. There will normally be a line that says `Author: Your Name <yourname@example.com>`. If the email address in that commit matches one that GitHub recognizes (e.g. the one linked to your GitHub account), then GitHub will display your avatar next to each of your commits.

Thank you so much :)

**Pull request merged**